test/utils.py

```diff
@@ -55,58 +53,155 @@ def execute(self) -> subprocess.CompletedProcess:
         return process
 
 
-class ArtifactsCollection:
-    def __init__(self, variant: str):
+class ArtifactsCollection(ABC):
+    @abstractmethod
+    def collect(self):
+        pass
+
+    def __iter__(self):
+        return iter(self.collect())
+
+    def assert_exists_in(self, directory: Path):

```

Comment:
We have this logic in the collection because different collections require different criteria for existence. The ComponentReportsCollection is a collection of directories but the existance relies on a file "index.html" in the report's directory.